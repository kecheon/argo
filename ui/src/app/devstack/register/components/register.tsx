import {Page} from 'argo-ui';
import * as React from 'react';
import './styles.css';

export class Register extends React.Component {
    public handleChange = (event: any) => {};
    public handleSubmit = (event: any) => {};
    public render() {
        return (
            <Page title='Register' toolbar={{breadcrumbs: [{title: 'Register'}]}}>
                <div className='argo-container'>
                    <div className='wrapper'>
                        <div className='form-wrapper'>
                            <form onSubmit={this.handleSubmit} noValidate={true}>
                                <div className='fullName'>
                                    <label htmlFor='fullName'>Full Name</label>
                                    <input type='text' name='fullName' onChange={this.handleChange} />
                                </div>
                                <div className='email'>
                                    <label htmlFor='email'>Email</label>
                                    <input type='email' name='email' onChange={this.handleChange} />
                                </div>
                                <div className='password'>
                                    <label htmlFor='password'>Password</label>
                                    <input type='password' name='password' onChange={this.handleChange} />
                                </div>
                                <div className='submit'>
                                    <button>Register Me</button>
                                </div>
                            </form>
                        </div>
                    </div>
                </div>
            </Page>
        );
    }
}
