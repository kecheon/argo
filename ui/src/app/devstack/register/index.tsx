import {Register} from './components/register';

export default {
  component: Register
}
