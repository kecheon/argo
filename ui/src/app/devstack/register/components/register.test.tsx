import '@testing-library/jest-dom/extend-expect';
import {render, screen} from '@testing-library/react';
import * as React from 'react';
import {Register} from './register';

xdescribe('Registration component', () => {
<<<<<<< HEAD
    it('should render', async () => {
        render(<Register />);
        const items = await screen.findAllByText('input');
        expect(items).not.toBeEmptyDOMElement();
=======
    it('should render', () => {
        const {container} = render(<Register />);
        expect(container.firstChild).toMatchSnapshot();
>>>>>>> temp
    });
});
