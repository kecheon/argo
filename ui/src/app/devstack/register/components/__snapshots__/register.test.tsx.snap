// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration component should render 1`] = `
<div
  class="page page--has-toolbar"
>
  <div
    class="page__top-bar"
  >
    <div>
      <div
        class="top-bar"
      >
        <div
          class="row"
        >
          <div
            class="columns top-bar__left-side"
          >
            <div
              class="top-bar__breadcrumbs"
            >
              <span
                class="top-bar__breadcrumbs-last-item"
              >
                Register
              </span>
            </div>
          </div>
          <div
            class="top-bar__title text-truncate top-bar__right-side"
          >
            Register
          </div>
        </div>
      </div>
      <div
        class="top-bar row"
      >
        <div
          class="columns small-9 top-bar__left-side"
        />
        <div
          class="columns small-3 top-bar__right-side"
        />
      </div>
    </div>
  </div>
  <div
    class="page__content-wrapper"
  >
    <div
      class="argo-container"
    >
      <div>
        <h3>
          New Tenant
        </h3>
        Create user(admin) and namespace for workflow
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
      <button
        class="btn btn-secondary"
        type="submit"
      >
        Cancel
      </button>
      <form
        class=""
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="formBasicUsername"
                >
                  User Name*
                </label>
                <input
                  class="form-control"
                  id="formBasicUsername"
                  placeholder="Enter username"
                  type="text"
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="formBasicDescription"
                >
                  Description
                </label>
                <textarea
                  class="form-control"
                  id="formBasicDescription"
                  placeholder="Enter description"
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="formBasicEmail"
                >
                  Description
                </label>
                <input
                  class="form-control"
                  id="formBasicEmail"
                  placeholder="Enter email"
                  type="email"
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="formBasicPassword1"
                >
                  Password
                </label>
                <input
                  class="form-control"
                  id="formBasicPassword1"
                  placeholder="Password"
                  type="password"
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="formBasicPassword2"
                >
                  Confirm Password
                </label>
                <input
                  class="form-control"
                  id="formBasicPassword2"
                  placeholder="Password again"
                  type="password"
                />
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="formBasicEmail"
                >
                  Namespace Name*
                </label>
                <input
                  class="form-control"
                  id="formBasicEmail"
                  placeholder="Enter namespace"
                  type="string"
                />
                <small
                  class="text-muted form-text"
                >
                  We'll never share your email with anyone else.
                </small>
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="formBasicDescriptionNamespace"
                >
                  Namespace Description
                </label>
                <textarea
                  class="form-control"
                  id="formBasicDescriptionNamespace"
                  placeholder="Enter namespace description"
                />
              </div>
              <label
                class="form-label"
              >
                Quota
              </label>
              <div
                class="input-group"
              >
                <input
                  class="form-control"
                  placeholder="CPU"
                />
                <div
                  class="input-group-append"
                >
                  <button
                    class="btn btn-outline-secondary"
                    type="button"
                  >
                    +
                  </button>
                  <button
                    class="btn btn-outline-secondary"
                    type="button"
                  >
                    -
                  </button>
                </div>
              </div>
              <div
                class="input-group"
              >
                <input
                  class="form-control"
                  placeholder="Memory"
                />
                <div
                  class="input-group-append"
                >
                  <button
                    class="btn btn-outline-secondary"
                    type="button"
                  >
                    +
                  </button>
                  <button
                    class="btn btn-outline-secondary"
                    type="button"
                  >
                    -
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
